app/src/main/java/com/realityexpander/tasky/agenda_feature/presentation/agenda_screen/AgendaViewModel.kt

```diff
@@ -122,18 +214,32 @@ class AgendaViewModel @Inject constructor(
 //                    )
 //                }
 //            }
-            is AgendaEvent.ToggleTaskDropdown -> {
+            is AgendaEvent.ShowAgendaItemDropdown -> {
                 _agendaState.update {
                     it.copy(
-                        isTaskDropdownShowing = !agendaState.value.isTaskDropdownShowing
+                        agendaItemIdForMenu = event.agendaItemId,
                     )
                 }
+                yield()
+            }
+            is AgendaEvent.ToggleTaskCompleted -> {
+                _agendaState.update {
+                    it.copy(
+                        agendaItems = it.agendaItems.map { agendaItem ->
+                            if (agendaItem.id == event.agendaItemId) {
+                                (agendaItem as AgendaItem.Task).copy(isDone = !agendaItem.isDone)
```

Comment:
Maybe add an additional safety-check here for `agendaItem is AgendaItem.Task` just to avoid this accidentally gets called for a non-Task

Reply:
OK I will add this.